File: HtmlPrototype/game/gameplay/gamecontext.js
function gameContext(gameboard, updateUiFunction) {
    $this = this;
    var $gameboard = gameboard;
    var $currentRound = 0;
    var $currentPlayerTurn = 0;
    var $thisTurn = null;
    var $currentPlayer = null;
    var $updateUiFunction = updateUiFunction;

    var $players = new Array();
    $players[1] = new player(1, "Player 1");
    $players[2] = new player(2, "Player 2");

    this.systems = function(){
        return $gameboard;
    };

    this.currentRound = function(){
        return $currentRound;
    };
    this.currentPlayer = function(){
        return $currentPlayer;
    };

    this.currentPlayerTurn = function(){
      return $currentPlayerTurn;
    };

    this.thisTurn = function(){
      return $thisTurn;
    };

    this.startTurn = function(){
        //commit previous turn
        if($thisTurn != null){
            $thisTurn.commitTurn();
        }

        //start next turn
        if($currentPlayerTurn == 1){
            $currentPlayerTurn = 2;
        }
        else if($currentPlayerTurn == 2){
            $currentPlayerTurn = 1;
            $currentRound++;
        }

        if($currentPlayerTurn == 0){
            $currentPlayerTurn = 1;
            $currentRound = 1;
        }

        $currentPlayer = $players[$currentPlayerTurn];

        $thisTurn = new turn($currentPlayer);
        $thisTurn.tick(this);


        $(".round-number").text($currentRound);

        this.updateUi();
    };

    this.getSelectedSystem = function(){
        return $('body').data('activeSystem');
    };

    this.setSelectedSystem = function(system){
         $('body').data('activeSystem', system);
    };

    this.queueAction = function(action){
        $thisTurn.actionQueue().push(action);

        this.updateUi();
    };

    this.updateUi = function(){
        $updateUiFunction($this);
        //$playerRenderer.render($currentPlayer);
        //$worldRenderer.render($uiElement);
    };


}

File: StarBastard/StarBastard.Core/Gameplay/GameContext.cs
using System;
using System.Collections.Generic;
using StarBastard.Core.Universe.Systems;

namespace StarBastard.Core.Gameplay
{
    public class GameContext
    {
        private readonly Action<GameContext> _render;
        private int currentRound = 0;
        private int _currentPlayerTurn = 0;
        private Turn _thisTurn = null;
        private Player _currentPlayer = null;

        public Dictionary<int, Player> Players { get; set; }
        public GameBoard Systems { get; private set; }

        public GameContext(GameBoard gameboard, Action<GameContext> render)
        {
            Systems = gameboard;
            _render = render;
            Players = new Dictionary<int, Player>
            {
                {1, new Player(1, "Player 1")},
                {2, new Player(2, "Player 2")}
            };
        }


        public void StartTurn()
        {
            if (_thisTurn != null)
            {
                _thisTurn.commitTurn();
            }

            //start next turn
            if (_currentPlayerTurn == 1)
            {
                _currentPlayerTurn = 2;
            }
            else if (_currentPlayerTurn == 2)
            {
                _currentPlayerTurn = 1;
                currentRound++;
            }

            if (_currentPlayerTurn == 0)
            {
                _currentPlayerTurn = 1;
                currentRound = 1;
            }

            _currentPlayer = Players[_currentPlayerTurn];

            _thisTurn = new Turn(_currentPlayer);
            _thisTurn.tick(this);


            _render(this);
        }
    }
}
